src/main/java/sirius/db/mixing/types/MultiLanguageString.java

```diff
@@ -0,0 +1,174 @@
+     * Creates a new multi-language string property without default language.
+     */
+    public MultiLanguageString() {
+        this(null);
+    }
+
+    /**
+     * Creates a new multi-language string property with default language.
+     *
+     * @param defaultLanguage the default language code
+     */
+    public MultiLanguageString(String defaultLanguage) {
+        this.defaultLanguage = defaultLanguage;
+    }
+
+    @Override
+    protected boolean valueNeedsCopy() {
+        return false;
+    }
+
+    @Override
+    protected String copyValue(String value) {
+        return value;
+    }
+
+    /**
+     * Adds a new text for the given language.
+     *
+     * @param language the language code
+     * @param text     the text associated with the language
+     * @return the object itself for fluent method calls
+     */
+    public MultiLanguageString addText(String language, String text) {
+        assertLanguage(language);
+        put(language, text);
+        return this;
+    }
+
+    /**
+     * Checks if a text exists for a given language.
+     *
+     * @param language the language code
+     * @return <tt>true</tt> when a text exists, otherwise <tt>false</tt>
+     */
+    public boolean hasText(String language) {
+        return data().containsKey(language);
+    }
+
+    /**
+     * Returns the text associated with the default language.
+     *
+     * @return the text
+     */
+    public String getRequiredText() {
+        return getRequiredText(defaultLanguage);
+    }
+
+    /**
+     * Returns the text associated with a given language.
+     *
+     * @param language the language code
+     * @return the text or an {@link sirius.kernel.health.HandledException} if not existent
+     */
+    public String getRequiredText(String language) {
+        if (!hasText(language)) {
+            throw Exceptions.createHandled()
+                            .withNLSKey("MultiLanguageString.textDoesNotExist")
+                            .set("language", language)
+                            .handle();
+        }
+        return fetchText(language);
+    }
+
+    /**
+     * Returns an optional text associated with the default language.
+     *
+     * @return an Optional String containing the text, otherwise an empty Optional
+     */
+    public Optional<String> getText() {
+        return getText(defaultLanguage);
+    }
+
+    /**
+     * Returns an optional text associated with a given language.
+     *
+     * @param language the language code
+     * @return an Optional String containing the text, otherwise an empty Optional
+     */
+    public Optional<String> getText(String language) {
+        if (!hasText(language)) {
+            return Optional.empty();
+        }
+        return Optional.of(fetchText(language));
+    }
+
+    /**
+     * Returns the text associated with the default language.
+     *
+     * @return the text if it exists, otherwise <tt>null</tt>
+     */
+    public String fetchText() {
```

Comment:
@Nullable

Reply:
Do we want to annotate all `fetchText` methods with a @nullable?
If so, basically any method that do not return void needs either nullable or nonnull

Reply:
we could argue about the nonnull for optionals as it is kind of a no-brainer
in this case i'd welcome the @nullable as then the compiler can enforce some checks